api/include/opentelemetry/trace/span_id.h

```diff
@@ -61,7 +61,8 @@ class SpanId final
   bool IsValid() const noexcept
   {
     static_assert(kSize == 8, "update is needed if kSize is not 8");
```

Comment:
This `static_assert` could be removed then as the code doesn't make such assumption anymore?

Reply:
fixed